src/orca/services/nextflowtower/models.py

```diff
@@ -28,83 +30,153 @@ class TaskStatus(Enum):
 
 
 @dataclass(kw_only=False)
-class User:
-    """Nextflow Tower user."""
+class BaseTowerModel:
+    """Base model for Nextflow Tower models.
 
-    id: int
-    username: str
-    email: str
+    Attributes:
+        _key_mapping: Mapping between Python and API key names.
+            Only discordant names need to be listed.
+    """
+
+    _: KW_ONLY
+    raw: Optional[dict[str, Any]] = field(default=None, repr=False, compare=False)
```

Comment:
> Does this mean each class has a .raw attribute with the full json response?

Yes, in case we need information from a response that isn't yet available through an attribute. Otherwise, you would have to issue a new py-orca release, which is annoying. 

> If so, can you speak to why you prefer this over creating kwargs? (e.g. class.non_key_map_1)

I'm not sure I understand the alternative you're proposing.

Reply:
Ah, I see. Here's my rationale for my approach:

- Some (most?) responses are pretty long and often nested. For example, here's the [response](https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/ccd3a5739be80c801d4737a6ab9bd39121055060/tests/services/nextflowtower/responses.py#L83-L169) when retrieving details on a compute environment. Exposing all of these values would be onerous, because each one has to be typed and mapped using `_key_mapping`. 
- I could override the `__getattr__()` method to fall back on the `raw` response if an instance attribute is missing, but I've had a hard time maintaining custom `__getattr__` logic in the past. Also, it would result in a mix of snake case and camel case. 
- We will generally only need a small subset of values in each HTTP response, so exposing all of them feels like premature optimization. YAGNI. 

If adding new attributes becomes burdensome, we can revisit this approach. I do appreciate you keeping me accountable and making sure that I've thought through my approach!

Reply:
Thanks for the thorough explanation - this makes sense to me!